docs/ratt-transform/index.md

```diff
@@ -675,84 +675,384 @@ app.use(
 
 Notice that it is almost never useful to store the empty string in linked data.  So the treatment of the empty string as a NULL value is the correct default behavior.
 
-<h2 id='access-nested-data'> Access data</h2>
+## Tree-shaped data
+
+Tree-shaped data is very common in different source systems.  For example, JSON- and XML-based formats are tree-shaped.  RATT has good support for working with such sources.
+
+We will use the following example in this section:
 
-### JSON
-It is often the case that we want to access data that are nested and use them to create linked data. For example, we want to access the types of the properties that a person named J.D. has.
 ```json
 {
-  "name": "J.D.",
-  "properties": [
+  "metadata": {
+    "title": {
+      "name": "Data about countries."
+    }
+  },
+  "data": {
+    "countries": [
+      {
+        "id": "nl",
+        "name": "The Netherlands"
+      },
+      {
+        "id": "de",
+        "name": "Germany"
+      }
+    ]
+  }
+}
+```
+
+Since the RATT record abstracts away the syntactic specifics of the source system format, the above example may be read from a JSON source (see [connecting JSON sources](#json)), or from another kind of source.  For example, the RATT record may be read from the following XML source, assuming the contents of the `<root>` tag are selected:
+
+```xml
+<?xml version="1.0"?>
+<root>
+  <metadata>
+    <title>
+      <value>Data about countries.</value>
+    </title>
+  </metadata>
+  <data>
+    <countries>
+      <id>nl</id>
+      <name>The Netherlands</name>
+    </countries>
+    <countries>
+      <id>de</id>
+      <name>Germany</name>
+    </countries>
+  </data>
+</root>
+```
+
+See [the section on connecting XML sources](#xml) for more information.
+
+### Specifying paths (nested keys)
+
+In tabular data, keys (or column names) are singular.  But in tree-shaped data a *path* of the tree can consist of one or more keys that must be traversed in sequence.
+
+Paths are specified as dot-separated sequences of keys, starting at the top-level and ending at the required value.  For the above example, RATT can access the `"name"` key inside the `"title"` key, which itself is nested inside the `"metadata"` key.  This path is expressed in [1].  Notice that the path expressed in [1] is different from the path expressed in [2], which also access the `"name"` key, but nested inside the `"countries"` and then `"data"` keys.
+
+```
+[1] metadata.title.name
+[2] data.countries.name
```

Comment:
the second path is not correct: `countries` has an array of objects, while `title` includes an object (which makes path [1]  correct). Thus,
[2] data.countries[x].name where x can be 0 or 1.

Reply:
Ah, you're right.  That breaks my example :-(